react/components/ListDetails/index.tsx

```diff
@@ -0,0 +1,201 @@
+}
+
+class ListDetail extends Component<ListDetailProps & InjectedIntlProps & WithApolloClient<{}>, ListDetailState> {
+  public state: ListDetailState = {
+    isLoading: true,
+    selectedItems: [],
+  }
+
+  private options: Option[] = [
+    {
+      onClick: () => this.setState({ showUpdateList: true }),
+      title: translate('wishlist-option-configuration', this.props.intl),
+    },
+    {
+      onClick: () => this.setState({ showDeleteConfirmation: true }),
+      title: translate('wishlist-option-delete', this.props.intl),
+    },
+  ]
+
+  public componentDidMount(): void {
+    const { listId, client } = this.props
+    if (client) {
+      getListDetailed(client, listId)
+        .then(response => {
+          this.setState({ list: response.data.list, isLoading: false })
+        })
+        .catch(err => console.error(err))
+    }
+  }
+
+  public render(): ReactNode {
+    const { list, isLoading, showDeleteConfirmation, showUpdateList } = this.state
+    const { intl, listId } = this.props
+    return (
+      <div className="fixed top-0 left-0 vw-100 vh-100 flex flex-column z-4 bg-base">
+        {isLoading ? renderLoading() : this.renderContent()}
+        {showDeleteConfirmation && (
+          <Dialog
+            message={`${translate('wishlist-delete-confirmation-message', intl)} "${list.name}"?`}
+            onClose={() => this.setState({ showDeleteConfirmation: false })}
+            onSuccess={this.handleDeleteList}
+          />
+        )}
+        {showUpdateList && (
+          <div className="fixed top-0 left-0 w-100 bg-base">
+            <UpdateList
+              onClose={() => this.setState({ showUpdateList: false })}
+              list={{ ...list, id: listId }}
+              onFinishUpdate={this.onFinishUpdate}
+            />
+          </div>
+        )}
+      </div>
+    )
+  }
+
+  private renderContent = (): ReactNode => {
+    const { list: { items, name }, selectedItems } = this.state
+    return (
+      <Fragment>
+        <Header
+          title={name}
+          onClose={this.handleOnClose}
+          showIconBack
+        >
+          <MenuOptions options={this.options} size={Size.large} />
+        </Header>
+        <Content
+          items={items}
+          onItemSelect={this.onItemSelectedChange}
+          onItemRemove={this.onItemRemove}
+        />
+        {items.length > 0 && (
+          <Footer
+            items={selectedItems}
+            onAddToCart={this.addItensToCart}
+          />
+        )}
+      </Fragment>
+    )
+  }
+
+  private createItemShapeFromItem = ({ product: { items } }: any) => {
+    const sku = items[0]
+    return {
+      id: Number(sku.itemId),
+      quantity: 1,
+      seller: Number(sku.sellers[0].sellerId),
+    }
+  }
+
+  private addItensToCart = (): Promise<any> => {
+    const { orderFormContext } = this.props
+    const { selectedItems } = this.state
+
+    this.setState({ isAddingToCart: true })
+    return orderFormContext
+      .addItem({
```

Comment:
You might want to talk with @estacioneto about the optimistic cart

Reply:
Actually, try using the `BuyButton` component here...

Reply:
That's a good idea.